src/dynimage_serde.rs

```diff
@@ -0,0 +1,182 @@
+use crate::{DynamicImage, ImageFormat};
+use base64::{engine::general_purpose::STANDARD_NO_PAD, Engine};
+use serde::{
+    de::{self, Visitor},
+    ser, Deserialize, Deserializer, Serialize,
+};
+use std::{fmt, io::{Cursor, Write}};
+
+impl<'de> Deserialize<'de> for DynamicImage {
```

Comment:
This feel like over complicating things. We could instead use serde-derive with a proxy struct:
```rust
mod serialize {
    /// Proxy struct for serializing a DynamicImage
    #[derive(Serialize, Deserialize)]
    struct DynamicImage(Vec<u8>);

    impl<'de> Deserialize<'de> for crate::DynamicImage {
        fn deserialize<D: Deserializer<'de>>(deserializer: D) -> Result<DynamicImage, D::Error> {
            let img = DynamicImage::deserialize(deserializer)?;
            crate::load_from_memory(&img.0).map_err(de::Error::custom)
        }
    }

    ...
}